nova_vm/src/ecmascript/builtins/array_buffer/abstract_operations.rs

```diff
@@ -366,39 +376,49 @@ pub(crate) fn get_raw_bytes_from_shared_block(
 /// integer), type (a TypedArray element type), isTypedArray (a Boolean),
 /// and order (SEQ-CST or UNORDERED) and optional argument isLittleEndian
 /// (a Boolean) and returns a Number or a BigInt.
-pub(crate) fn get_value_from_buffer(
-    _array_buffer: ArrayBuffer,
-    _byte_index: u32,
-    _type: (),
+pub(crate) fn get_value_from_buffer<T: Viewable>(
+    agent: &mut Agent,
+    array_buffer: ArrayBuffer,
+    byte_index: usize,
     _is_typed_array: bool,
     _order: Ordering,
-    _is_little_endian: Option<bool>,
-) {
+    is_little_endian: Option<bool>,
+) -> Value {
     // 1. Assert: IsDetachedBuffer(arrayBuffer) is false.
+    debug_assert!(!array_buffer.is_detached(agent));
     // 2. Assert: There are sufficient bytes in arrayBuffer starting at byteIndex to represent a value of type.
-    // 3. Let block be arrayBuffer.[[ArrayBufferData]].
     // 4. Let elementSize be the Element Size value specified in Table 71 for Element Type type.
+    // 3. Let block be arrayBuffer.[[ArrayBufferData]].
+    let block = agent[array_buffer].get_data_block();
     // 5. If IsSharedArrayBuffer(arrayBuffer) is true, then
     // a. Assert: block is a Shared Data Block.
     // b. Let rawValue be GetRawBytesFromSharedBlock(block, byteIndex, type, isTypedArray, order).
     // 6. Else,
     // a. Let rawValue be a List whose elements are bytes from block at indices in the interval from byteIndex (inclusive) to byteIndex + elementSize (exclusive).
     // 7. Assert: The number of elements in rawValue is elementSize.
+    // TODO: Add [[LittleEndian]] to the agent record.
```

Comment:
**note:** The LE status of the Agent record depends on just the machine's natural LE status, so this should instead be a build config thing basically. On BE build targets it'll be BE but otherwise LE.